lib/wasix/src/state/linker.rs

```diff
@@ -3174,11 +3200,20 @@ impl InstanceGroupState {
                             ));
                         }
                         trace!(?module_handle, ?import, "setjmp/longjmp exception tag");
-                        imports.define(
-                            import.module(),
-                            import.name(),
-                            Tag::new(store, vec![Type::I32]),
-                        );
+                        imports.define(import.module(), import.name(), self.c_longjmp.clone());
+                        continue;
+                    }
+                    "__cpp_exception" => {
+                        if !matches!(import.ty(), ExternType::Tag(ty) if *ty.params == [Type::I32])
+                        {
+                            return Err(LinkError::BadImport(
+                                import.module().to_string(),
+                                import.name().to_string(),
+                                import.ty().clone(),
+                            ));
+                        }
+                        trace!(?module_handle, ?import, "C++ exception tag");
+                        imports.define(import.module(), import.name(), self.cpp_exception.clone());
```

Comment:
This is the exact same as above. Is there a reason for not having this in a shared function?

Reply:
Yes, because we were initially lazy XD Let's do some refactoring on the Linker at some point, I'm sure there's more duplicate code we can eliminate.